test/e2e_node/docker_test.go

```diff
@@ -70,4 +77,174 @@ var _ = framework.KubeDescribe("Docker features [Feature:Docker]", func() {
+			framework.ExpectNoError(err)
+			if !isSupported {
+				framework.Skipf("Docker live-restore is not supported.")
+			}
+
+			By("Check whether live-restore is enabled.")
+			isEnabled, err := isDockerLiveRestoreEnabled()
+			framework.ExpectNoError(err)
+			framework.Logf("Docker live-restore enabled: %t", isEnabled)
+			if !isEnabled {
+				// Enables Docker live-restore if it's supported but disabled.
+				By("Enable Docker live-restore.")
+				framework.ExpectNoError(setDockerLiveRestore(true))
+
+				b, err := isDockerLiveRestoreEnabled()
+				framework.ExpectNoError(err)
+				if !b {
+					framework.Logf("Failed to enable Docker live-restore: %v", err)
+				}
+
+				defer func() {
+					By("Disable Docker live-restore.")
+					time.Sleep(2 * time.Second)
+					framework.ExpectNoError(setDockerLiveRestore(false))
+				}()
+			}
+
+			// Creates a temporary directory that will be mounted into the
+			// container, serving as the communication channel between the host
+			// and the container.
+			By("Create temporary directory for mount.")
+			tempDir, err := ioutil.TempDir("", "")
+			framework.ExpectNoError(err)
+			defer func() {
+				By("Remove temporary directory.")
+				defer os.RemoveAll(tempDir)
+			}()
+
+			// Creates a container that
+			//   - writes the current timestamp every second to the
+			//     current-timestamp file, and
+			//   - writes the start timestamp to the start-timestamp file
+			// in the directory that's mounted from the host machine.
+			//
+			// We will be able to tell
+			//   - whether the container is running by checking if the
+			//     current-timestamp increases, and
+			//   - whether the container has been restarted by checking if the
+			//     start-timestamp has changed.
+			cmd := "" +
+				// Writes the start time on start.
+				"date +%s > /test-dir/" + startTimeFilename + "; " +
+				// Writes the current timestamp every second.
+				"while true; do " +
+				"    date +%s > /test-dir/" + currentTimeFilename + "; " +
+				"    sleep 1; " +
+				"done"
+			By("Create the test pod.")
+			f.PodClient().CreateSync(&v1.Pod{
+				ObjectMeta: metav1.ObjectMeta{Name: "live-restore-test-pod"},
+				Spec: v1.PodSpec{
+					Containers: []v1.Container{{
+						Name:    "live-restore-test-container",
+						Image:   "gcr.io/google_containers/busybox:1.24",
+						Command: []string{"/bin/sh"},
+						Args:    []string{"-c", cmd},
+						VolumeMounts: []v1.VolumeMount{
+							{
+								Name:      "live-restore-test-volume",
+								MountPath: "/test-dir",
+							},
+						},
+					}},
+					Volumes: []v1.Volume{{
+						Name: "live-restore-test-volume",
+						VolumeSource: v1.VolumeSource{
+							HostPath: &v1.HostPathVolumeSource{Path: tempDir},
+						},
+					}},
+				},
+			})
+
+			startTime1, err := getTimestamp(filepath.Join(tempDir, startTimeFilename))
+			framework.ExpectNoError(err)
+
+			By("Stop Docker daemon.")
+			framework.ExpectNoError(stopDockerDaemon())
```

Comment:
I'm not sure whether we currently run the node e2e's in parallel, but this probably shouldn't run in parallel (is there a way to tell the framework?)